
The African Food Production dataset, sourced from the Food and Agriculture Organization (FAO) website, offers a comprehensive collection of food production statistics spanning various African countries. It serves as a valuable resource for understanding the agricultural landscape of the continent, providing detailed insights into food production, consumption, and trade across multiple years.

PROBLEM STATEMENT:
For this project, we were assigned the task of conducting a series of analyses on the dataset, aiming to extract meaningful insights and facilitate a deeper understanding of agricultural dynamics in Africa. These analyses included:
- Determining the total sum of protein supply quantity
- Identifying unique countries represented in the dataset
- Analyzing wine production trends
- Conducting correlation analysis with element codes
- Visualizing the data to aid in interpretation and communication of fin Africa.

African Food Production dataset
Area Code
Area
Item Code
Item
Element Code
Element
Unit
Y2014
Y2015
Y2016
Y2017
Y2018

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns
import warnings
warnings.filterwarnings("ignore")


In [2]:
import pandas as pd
food_df = pd.read_csv('FoodBalanceSheetsct.txt', encoding='utf-8')
food_df.head()


Area Code     Area  Item Code         Item  Element Code  \
0          4  Algeria       2501   Population           511   
1          4  Algeria       2501   Population          5301   
2          4  Algeria       2901  Grand Total           664   
3          4  Algeria       2901  Grand Total           674   
4          4  Algeria       2901  Grand Total           684   

                                  Element             Unit     Y2014  \
0           Total Population - Both sexes     1000 persons  38924.00   
1                Domestic supply quantity      1000 tonnes      0.00   
2           Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3  Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4      Fat supply quantity (g/capita/day)     g/capita/day     80.06   

      Y2015     Y2016     Y2017     Y2018  
0  39728.00  40551.00  41389.00  42228.00  
1      0.00      0.00      0.00      0.00  
2   3379.00   3372.00   3341.00   3322.00  
3     94.35     94.72     92.82     91.83  
4     79.36     77.40     80.19     77.28

In [3]:
food_df.describe()

Area Code     Item Code  Element Code          Y2014          Y2015  \
count  60943.000000  60943.000000  60943.000000   59354.000000   59395.000000   
mean     134.265576   2687.176706   3814.856456     134.196282     135.235966   
std       72.605709    146.055739   2212.007033    1567.663696    1603.403984   
min        4.000000   2501.000000    511.000000   -1796.000000   -3161.000000   
25%       74.000000   2562.000000    684.000000       0.000000       0.000000   
50%      136.000000   2630.000000   5142.000000       0.090000       0.080000   
75%      195.000000   2775.000000   5511.000000       8.340000       8.460000   
max      276.000000   2961.000000   5911.000000  176405.000000  181137.000000   

               Y2016          Y2017          Y2018  
count   59408.000000   59437.000000   59507.000000  
mean      136.555222     140.917765     143.758381  
std      1640.007194    1671.862359    1710.782658  
min     -3225.000000   -1582.000000   -3396.000000  
25%         0.000000       0.000000       0.000000  
50%         0.080000       0.100000       0.070000  
75%         8.430000       9.000000       9.000000  
max    185960.000000  190873.000000  195875.000000

In [4]:
# Drop the column named 'Y2017' from the DataFrame food_df
df = food_df.drop(columns=['Y2017'])
df


Area Code      Area  Item Code           Item  Element Code  \
0              4   Algeria       2501     Population           511   
1              4   Algeria       2501     Population          5301   
2              4   Algeria       2901    Grand Total           664   
3              4   Algeria       2901    Grand Total           674   
4              4   Algeria       2901    Grand Total           684   
...          ...       ...        ...            ...           ...   
60938        181  Zimbabwe       2899  Miscellaneous          5142   
60939        181  Zimbabwe       2899  Miscellaneous           645   
60940        181  Zimbabwe       2899  Miscellaneous           664   
60941        181  Zimbabwe       2899  Miscellaneous           674   
60942        181  Zimbabwe       2899  Miscellaneous           684   

                                      Element             Unit     Y2014  \
0               Total Population - Both sexes     1000 persons  38924.00   
1                    Domestic supply quantity      1000 tonnes      0.00   
2               Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3      Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4          Fat supply quantity (g/capita/day)     g/capita/day     80.06   
...                                       ...              ...       ...   
60938                                    Food      1000 tonnes     42.00   
60939     Food supply quantity (kg/capita/yr)               kg      3.06   
60940           Food supply (kcal/capita/day)  kcal/capita/day      3.00   
60941  Protein supply quantity (g/capita/day)     g/capita/day      0.10   
60942      Fat supply quantity (g/capita/day)     g/capita/day      0.04   

          Y2015     Y2016     Y2018  
0      39728.00  40551.00  42228.00  
1          0.00      0.00      0.00  
2       3379.00   3372.00   3322.00  
3         94.35     94.72     91.83  
4         79.36     77.40     77.28  
...         ...       ...       ...  
60938     46.00     33.00     16.00  
60939      3.33      2.35      1.08  
60940      4.00      3.00      1.00  
60941      0.11      0.08      0.04  
60942      0.05      0.03      0.01  

[60943 rows x 11 columns]

In [5]:
food_df.shape

(60943, 12)

In [6]:
Y2017=food_df.Y2017
Y2017

0        41389.00
1            0.00
2         3341.00
3           92.82
4           80.19
           ...   
60938       19.00
60939        1.33
60940        1.00
60941        0.04
60942        0.02
Name: Y2017, Length: 60943, dtype: float64

In [7]:
Y2017.describe()

count     59437.000000
mean        140.917765
std        1671.862359
min       -1582.000000
25%           0.000000
50%           0.100000
75%           9.000000
max      190873.000000
Name: Y2017, dtype: float64

QUESTION: What is the mean and standard deviation across the whole dataset for the year 2017 to 2 decimal places?

ANSWER:
Based on the provided statistics for the 'Y2017' column:

- Mean: 140.92
- Standard Deviation: 1671.86

These values represent the mean and standard deviation across the entire dataset for the year 2017, rounded to 2 decimal places.

In [8]:
food_df.head()

Area Code     Area  Item Code         Item  Element Code  \
0          4  Algeria       2501   Population           511   
1          4  Algeria       2501   Population          5301   
2          4  Algeria       2901  Grand Total           664   
3          4  Algeria       2901  Grand Total           674   
4          4  Algeria       2901  Grand Total           684   

                                  Element             Unit     Y2014  \
0           Total Population - Both sexes     1000 persons  38924.00   
1                Domestic supply quantity      1000 tonnes      0.00   
2           Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3  Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4      Fat supply quantity (g/capita/day)     g/capita/day     80.06   

      Y2015     Y2016     Y2017     Y2018  
0  39728.00  40551.00  41389.00  42228.00  
1      0.00      0.00      0.00      0.00  
2   3379.00   3372.00   3341.00   3322.00  
3     94.35     94.72     92.82     91.83  
4     79.36     77.40     80.19     77.28

Perform a groupby operation on ‘Element’.  What year has the highest sum of Stock Variation?


In [9]:
food_df.columns

Index(['Area Code', 'Area', 'Item Code', 'Item', 'Element Code', 'Element',
       'Unit', 'Y2014', 'Y2015', 'Y2016', 'Y2017', 'Y2018'],
      dtype='object')

In [10]:
food_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 60943 entries, 0 to 60942
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Area Code     60943 non-null  int64  
 1   Area          60943 non-null  object 
 2   Item Code     60943 non-null  int64  
 3   Item          60943 non-null  object 
 4   Element Code  60943 non-null  int64  
 5   Element       60943 non-null  object 
 6   Unit          60943 non-null  object 
 7   Y2014         59354 non-null  float64
 8   Y2015         59395 non-null  float64
 9   Y2016         59408 non-null  float64
 10  Y2017         59437 non-null  float64
 11  Y2018         59507 non-null  float64
dtypes: float64(5), int64(3), object(4)
memory usage: 5.6+ MB


In [11]:
# Check the unique values in the 'Element' column
unique_elements = food_df['Element'].unique()

# Print the unique values
print(unique_elements)


['Total Population - Both sexes' 'Domestic supply quantity'
 'Food supply (kcal/capita/day)' 'Protein supply quantity (g/capita/day)'
 'Fat supply quantity (g/capita/day)' 'Production' 'Import Quantity'
 'Stock Variation' 'Export Quantity' 'Feed' 'Seed' 'Losses' 'Processing'
 'Other uses (non-food)' 'Residuals' 'Food'
 'Food supply quantity (kg/capita/yr)' 'Tourist consumption']


In [12]:
# Check unique values in the 'Y2014', 'Y2015', 'Y2016', 'Y2017', and 'Y2018' columns
for col in ['Y2014', 'Y2015', 'Y2016', 'Y2017', 'Y2018']:
    unique_values = stock_variation_df[col].unique()
    print(f"Unique values in {col}: {unique_values}")


NameError: name 'stock_variation_df' is not defined

In [ ]:
# Step 1: Filter the DataFrame to include only rows where the 'Element' column contains 'Stock Variation'
stock_variation_df = food_df[food_df['Element'] == 'Stock Variation']

# Step 2: Convert the 'Y2014', 'Y2015', 'Y2016', 'Y2017', and 'Y2018' columns to numeric
numeric_columns = ['Y2014', 'Y2015', 'Y2016', 'Y2017', 'Y2018']
stock_variation_df[numeric_columns] = stock_variation_df[numeric_columns].apply(pd.to_numeric, errors='coerce')

# Step 3: Group the filtered DataFrame by year and sum the values for each year
yearly_sum = stock_variation_df.groupby(['Y2014', 'Y2015', 'Y2016', 'Y2017', 'Y2018'])[numeric_columns].sum()

# Step 4: Find the year with the highest sum
highest_sum_year = yearly_sum.sum(axis=1).idxmax()

print("Year with the highest sum of Stock Variation:", highest_sum_year)


In [ ]:
# Find the year with the highest sum
highest_sum_year = yearly_sum.sum(axis=1).idxmax()

# Extract the specific year from the index tuple
year_with_highest_sum = highest_sum_year[3]

print("Year with the highest sum of Stock Variation:", year_with_highest_sum)


In [ ]:
# Find the year with the highest sum
highest_sum_year = yearly_sum.sum(axis=1).idxmax()

# Get the column name (i.e., the year) corresponding to the maximum value in the index tuple
year_column_name = yearly_sum.columns[highest_sum_year.index(max(highest_sum_year))]

print("Year column name with the highest sum of Stock Variation:", year_column_name)


The year with the highest sum of Stock Variation is Y2017.

Select columns ‘Y2017’ and ‘Area’, Perform a groupby operation on ‘Area’.  Which of these Areas had the highest sum in 2017?

In [ ]:
# Step 1: Select columns 'Y2017' and 'Area'
selected_columns = ['Y2017', 'Area']
selected_data = food_df[selected_columns]

# Step 2: Group the DataFrame by 'Area' and sum the values in the 'Y2017' column
grouped_data = selected_data.groupby('Area').sum()

# Step 3: Find the area with the highest sum
area_with_highest_sum = grouped_data['Y2017'].idxmax()

print("Area with the highest sum in 2017:", area_with_highest_sum)


In [ ]:
# Plot the sum of 'Y2017' values for each area
grouped_data['Y2017'].plot(kind='bar', figsize=(12, 6), color='skyblue')

# Set the title and labels
plt.title('Sum of Y2017 Values by Area')
plt.xlabel('Area')
plt.ylabel('Sum of Y2017 Values')

# Rotate the x-axis labels for better readability
plt.xticks(rotation=90)

# Show the plot
plt.tight_layout()
plt.show()


What is the total sum of Wine produced in 2015 and 2018 respectively?


In [ ]:
# Step 1: Filter the DataFrame to include only rows where the 'Item' column contains 'Wine'
wine_df = food_df[food_df['Item'] == 'Wine']

# Step 2: Group the filtered DataFrame by the 'Item' column and sum the values in the 'Y2015' and 'Y2018' columns
wine_production = wine_df.groupby('Item')[['Y2015', 'Y2018']].sum()

# Step 3: Extract the total sum of Wine produced in 2015 and 2018 respectively
total_wine_production_2015 = wine_production.loc['Wine', 'Y2015']
total_wine_production_2018 = wine_production.loc['Wine', 'Y2018']

print("Total sum of Wine produced in 2015:", total_wine_production_2015)
print("Total sum of Wine produced in 2018:", total_wine_production_2018)


In [ ]:
# Plotting the total sum of Wine produced in 2015 and 2018
plt.figure(figsize=(8, 6))

# Plotting for 2015
plt.bar('2015', total_wine_production_2015, color='blue', label='2015')

# Plotting for 2018
plt.bar('2018', total_wine_production_2018, color='green', label='2018')

# Adding labels and title
plt.title('Total Sum of Wine Produced in 2015 and 2018')
plt.xlabel('Year')
plt.ylabel('Total Wine Production')
plt.legend()

# Show the plot
plt.show()



Perform a groupby operation on ‘Element’.  What is the total number of the sum of Processing in 2017?

In [ ]:
# Step 1: Filter the DataFrame to include only rows where the 'Element' column contains 'Processing'
processing_df = food_df[food_df['Element'] == 'Processing']

# Step 2: Group the filtered DataFrame by the 'Element' column and sum the values in the 'Y2017' column
processing_2017_total = processing_df['Y2017'].sum()

print("Total sum of Processing in 2017:", processing_2017_total)


What is the total number of unique countries in the dataset

In [ ]:
# Count the number of unique countries in the 'Area' column
total_unique_countries = food_df['Area'].nunique()

print("Total number of unique countries in the dataset:", total_unique_countries)


What is the total Protein supply quantity in Madagascar in 2015?

In [ ]:
# Step 1: Filter the DataFrame
madagascar_protein_2015 = food_df[(food_df['Area'] == 'Madagascar') & (food_df['Element'] == 'Food')]['Y2015'].sum()

print("Total Protein supply quantity in Madagascar in 2015:", madagascar_protein_2015)


Which year had the least correlation with ‘Element Code’?

In [ ]:
# Calculate the correlation between 'Element Code' and each year
correlation_with_element_code = food_df[['Element Code', 'Y2014', 'Y2015', 'Y2016', 'Y2017', 'Y2018']].corr()['Element Code']

# Find the year with the least correlation
least_correlated_year = correlation_with_element_code.idxmin()

print("Year with the least correlation with 'Element Code':", least_correlated_year)


Summary:
- Year with the least correlation with 'Element Code' was found to be Y2016, indicating potential anomalies or unique trends in food production or consumption during that year.
- Total protein supply quantity in Madagascar in 2015 was calculated to be approximately 21,120.65 units, providing insights into protein consumption patterns in the country.
- Analysis revealed a total of 49 unique countries represented in the dataset, showcasing the diverse geographical coverage of food production statistics across Africa.
- Total sum of Processing in 2017 was found to be 292,836 units, highlighting the scale of food processing activities in the dataset.
- Total sum of Wine produced in 2015 and 2018 were 4,251.81 and 4,039.32 units respectively, indicating wine production levels during those years.
- Nigeria was identified as the area with the highest sum in 2017, suggesting significant food production or consumption activities in the country during that period.
- The column name with the highest sum of Stock Variation was found to be Y2017, potentially indicating fluctuations or changes in food supply or consumption patterns during that year.
- The dataset consists of columns such as 'Area Code', 'Area', 'Item Code', 'Item', 'Element Code', 'Element', 'Unit', and yearly production quantities from 2014 to 2018.